server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java

```diff
@@ -174,127 +183,27 @@ public IssueQuery createFromRequest(Request request) {
     }
   }
 
-  private void addComponentParameters(IssueQuery.Builder builder, DbSession session,
-    @Nullable Boolean onComponentOnly,
+  private boolean mergeComponentParameters(DbSession session, Boolean onComponentOnly,
```

Comment:
As the goal of this method is to manage backward compatibility, please remove its name to make it clear what is the goal of this method